### PR TITLE
set REACT_APP _API_ENDPOINT and _AUTHENTICATION_URL as props

### PR DESCRIPTION
**Pull request by conorriches**
## Issue
Closes https://github.com/RaspberryPiFoundation/learner-experience-integration-tests/issues/5

## Description

* Changed how both utils function:
  * `src/utils/apiCallHandler.js`
  * `src/utils/userManager.js`
* I've considered how best to adapt these helpers - I could have gone with classes or functions, and while I considered changing these to be classes, this would require more verbose invocation throughout the codebase which ended up making code harder to read, whereas the function-based approach means most calls can be left as-is.
* `public/index.html` added which provides a handy link to the web-component.html rather than just a directory listing
* Removed a load of code that isn't used any more

### CODE REVIEW

File: src/utils/userManager.js [hunk 1]
```diff
@@ -5,13 +5,13 @@ const host = `${window.location.protocol}//${window.location.hostname}${
   window.location.port ? `:${window.location.port}` : ""
 }`;
 
-const userManagerConfig = {
+const userManagerConfig = ({ reactAppAuthenticationUrl }) => ({
```
Comment (sra405) [inline src/utils/userManager.js:8]: ❓ I'm struggling to see where `reactAppAuthenticationUrl` is set 🤔 I'm sure its just me but I can't see it 😅 or do we not need it?

File: src/redux/EditorSlice.test.js [hunk 2]
```diff
@@ -16,14 +7,25 @@ import reducer, {
...
+const mockReadProject = jest.fn();
+const mockCreateOrUpdateProject = jest.fn();
+
+jest.mock("../utils/apiCallHandler", () => () => ({
+  createRemix: jest.fn(mockCreateRemix),
+  deleteProject: jest.fn(mockDeleteProject),
+  loadAssets: jest.fn(mockLoadAssets),
+  readProject: jest.fn(mockReadProject),
+  createOrUpdateProject: jest.fn(mockCreateOrUpdateProject),
+}));
```
Comment (sra405) [inline src/redux/EditorSlice.test.js:28]: ❤️ this actually makes testing below read nicer imo

File: src/utils/userManager.js (hunk 1 above)
Comment (conorriches) [reply src/utils/userManager.js:8]: Yeah I had the same realisation when doing this 😅 
It's actually not yet used - only in `src/app/store.js` where I've left a TODO:

```
// TODO - not used but keeping this in preparation for using
// src/components/Editor/ImageUploadButton/ImageUploadButton.jsx
```

File: src/containers/WebComponentLoader.jsx [hunk 3]
```diff
@@ -31,24 +32,25 @@ const WebComponentLoader = (props) => {
...
-    showSavePrompt = false,
+    identifier,
+    instructions,
     loadRemixDisabled = false,
-    readOnly = false,
     outputOnly = false,
+    outputPanels = ["text", "visual"],
     outputSplitView = false,
+    projectNameEditable = false,
+    reactAppApiEndpoint = process.env.REACT_APP_API_ENDPOINT,
```
Comment (sra405) [inline src/containers/WebComponentLoader.jsx:45]: 💪 nice!! So given this is a default fallback do you reckon its worth keeping this env var in the `.env.example` and CI/CD so it'll work as is but can be overridden in the `editor-wc` props?

File: src/utils/userManager.js (hunk 1 above)
Comment (sra405) [reply src/utils/userManager.js:8]: that's mad that's the only use of that variable 😅

File: src/containers/WebComponentLoader.jsx [hunk 4]
```diff
@@ -135,6 +138,10 @@ const WebComponentLoader = (props) => {
     saveTriggered,
   });
 
+  useEffect(() => {
+    dispatch(setReactAppApiEndpoint(reactAppApiEndpoint));
+  }, [reactAppApiEndpoint, dispatch]);
```
Comment (sra405) [inline src/containers/WebComponentLoader.jsx:143]: 🔧 definitely could do with a test for this in the `WebComponentLoader.test.js` 👍

File: src/containers/WebComponentLoader.jsx (hunk 3 above)
Comment (conorriches) [reply src/containers/WebComponentLoader.jsx:45]: Sure! done in [727da9c](https://github.com/RaspberryPiFoundation/editor-ui/pull/1124/commits/727da9c16abb1638406c76afbf58544dab6ddbe8) :)

File: src/web-component.html [hunk 5]
```diff
@@ -43,6 +43,7 @@
       // sidebar
       webComp.setAttribute("with_projectbar", "true");
       webComp.setAttribute("with_sidebar", "true");
+      webComp.setAttribute("react_app_api_endpoint", "https://test-editor-api.raspberrypi.org");
```
Comment (sra405) [inline src/web-component.html:46]: 🔧  I'd be tempted removing this from the test page and letting it get set in CI still
```suggestion
```

**Review by sra405 — APPROVED** [review]
💪 this is awesome!

**Pull request merged**